src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -972,16 +972,29 @@ def customizePerJob(self, job):
 
         # Performance estimates
         if job.get('estimatedJobTime'):
-            jdl.append('+MaxWallTimeMins = %d\n' % int(job['estimatedJobTime'] / 60.0))
+            estimatedMins = int(job['estimatedJobTime'] / 60.0)
+            origCores = int(job.get('numberOfCores', 1))
+            estimatedMinsSingleCore = estimatedMins*origCores
+            jdl.append('+MaxWallTimeMins = WMCore_ResizeJob ? (%d/RequestCpus + 60) : %d\n' % (estimatedMinsSingleCore, estimatedMins))
+            
         if job.get('estimatedMemoryUsage'):
-            jdl.append('request_memory = %d\n' % int(job['estimatedMemoryUsage']))
+            extraMem = int(job.get('extraMemoryPerCore', 500))
+            baseMem = int(job['estimatedMemoryUsage'])
+            jdl.append('+RequestMemory = WMCore_ResizeJob ? (%d + RequestCpus*%d) : %d\n' % (baseMem, extraMem, baseMem))
+            
         if job.get('estimatedDiskUsage'):
             jdl.append('request_disk = %d\n' % int(job['estimatedDiskUsage']))
 
         # Set up JDL for multithreaded jobs
-        if job.get('numberOfCores', 1) > 1:
-            jdl.append('machine_count = 1\n')
-            jdl.append('request_cpus = %s\n' % job.get('numberOfCores', 1))
+        jdl.append('machine_count = 1\n')
+        minCores = int(job.get('minCores', 1))
+        maxCores = int(job.get('maxCores', 4))
+        origCores = int(job.get('numberOfCores', 1))
+        jdl.append('+MinCores = %d\n' % minCores)
+        jdl.append('+MaxCores = %d\n' % maxCores)
+        jdl.append('+RequestCpus = WMCore_ResizeJob ? (isUndefined(Cpus) ? MinCores : ((Cpus>MaxCores) ? MaxCores : Cpus)) : %d\n' % origCores)
```

Comment:
this would be a good place for a comment explaining what's going on for those not indoctrinated to the HTCondor high priesthood.